contracts/lib/FulfillmentApplier.sol

```diff
@@ -225,262 +234,183 @@ contract FulfillmentApplier is FulfillmentApplicationErrors {
-                // Increment the pointer to the fulfillment head by one word.
+            } {
+                // Increment position in considerationComponents head.
                 fulfillmentHeadPtr := add(fulfillmentHeadPtr, OneWord)
 
-                // Get the order index using the fulfillment pointer.
-                orderIndex := mload(mload(fulfillmentHeadPtr))
+                // Retrieve the order index using the fulfillment pointer.
+                let orderIndex := mload(mload(fulfillmentHeadPtr))
 
-                // Ensure the order index is in range.
+                // Ensure that the order index is not out of range.
                 if iszero(lt(orderIndex, mload(advancedOrders))) {
-                  throwInvalidFulfillmentComponentData()
+                    throwInvalidFulfillmentComponentData()
                 }
 
-                // Get pointer to AdvancedOrder element.
-                orderPtr := mload(
-                    add(
-                        add(advancedOrders, OneWord),
-                        mul(orderIndex, OneWord)
-                    )
+                // Read advancedOrders[orderIndex] pointer from its array head.
+                let orderPtr := mload(
+                    // Calculate head position of advancedOrders[orderIndex].
+                    add(add(advancedOrders, OneWord), mul(orderIndex, OneWord))
                 )
 
-                // Only continue if numerator is not zero.
-                if iszero(mload(
-                    add(orderPtr, AdvancedOrder_numerator_offset)
-                )) {
-                  continue
-                }
-
                 // Read the pointer to OrderParameters from the AdvancedOrder.
-                paramsPtr := mload(orderPtr)
+                let paramsPtr := mload(orderPtr)
 
-                // Load offer array pointer.
-                offerArrPtr := mload(
-                    add(
-                        paramsPtr,
-                        OrderParameters_offer_head_offset
-                    )
+                // Load the offer array pointer.
+                let offerArrPtr := mload(
+                    add(paramsPtr, OrderParameters_offer_head_offset)
                 )
 
-                // Get the item index using the fulfillment pointer.
-                itemIndex := mload(add(mload(fulfillmentHeadPtr), Fulfillment_itemIndex_offset))
+                // Retrieve item index using an offset of the fulfillment pointer.
+                let itemIndex := mload(
+                    add(mload(fulfillmentHeadPtr), Fulfillment_itemIndex_offset)
+                )
 
-                // Throw if itemIndex is out of the range of array.
-                if iszero(
-                    lt(itemIndex, mload(offerArrPtr))
+                // If the offer item index is out of range or the numerator
+                // is zero, skip this item.
+                if or(
+                    iszero(lt(itemIndex, mload(offerArrPtr))),
```

Comment:
Was there a particular reason, we also choose to skip OORs for offer array item index? Orders of `CONTRACT` order type cannot reduce the length of this array.